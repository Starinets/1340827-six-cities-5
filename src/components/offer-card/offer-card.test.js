import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import OfferCard from './offer-card';
import {
  emptyFunction,
  offer,
  OfferPlace,
  OfferCardImage
} from '../../test-data/test-data';

it(`Component rendered correctly -> OfferCard`, () => {
  const renderer = new ShallowRenderer();
  const tree = renderer.render(
      <OfferCard
        offer={ offer }
        offerPlace={ OfferPlace.NEIGHBORHOOD }
        imageSize={ OfferCardImage.List }
        onMouseOver={ emptyFunction }
      />
  );
  expect(tree).toMatchSnapshot();
});
