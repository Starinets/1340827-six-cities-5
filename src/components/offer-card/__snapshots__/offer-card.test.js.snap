// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component rendered correctly -> OfferCard 1`] = `
<article
  className="near-places__card place-card"
  onClick={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="near-places__image-wrapper place-card__image-wrapper"
  >
    <NavLink
      to="/offer/1"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height={200}
        src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg"
        width={260}
      />
    </NavLink>
  </div>
  <div
    className="place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          493
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <Memo(Connect(FavoriteButton))
        buttonHeight={19}
        buttonWidth={18}
        className="place-card"
        id={1}
        isFavorite={false}
      />
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "60%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <NavLink
        to="/offer/1"
      >
        The house among olive 
      </NavLink>
    </h2>
    <p
      className="place-card__type"
    >
      house
    </p>
  </div>
</article>
`;
